librosa/core/spectrum.py

```diff
@@ -1262,7 +1262,9 @@ def fmt(y, t_min=0.5, n_fmt=None, kind='cubic', beta=0.5, over_sample=1, axis=-1
         x_exp = np.clip(x_exp, float(t_min) / n, x[-1])
 
     # Make sure that all sample points are unique
-    assert len(np.unique(x_exp)) == len(x_exp)
+    # This should never happen!
+    if len(np.unique(x_exp)) != len(x_exp):
+        raise RuntimeError('Redundant sample positions in FMT')
```

Comment:
spell out "fast Mellin transform"
FMT is too uncommon of a TLA to be used

what does "this should never happen" mean here? is it on the responsability of the user or the programmer to have this not happen?

Reply:
I would say it's on the programmer.

To reconstruct what happened in the development of this function, there was an issue that the interpolator would sometimes fail (throw an exception) if the interpolated sample positions landed out of bounds from the observed samples.  This, I think, was only occurring in strange corner cases due to rounding error.  We hack around this by clamping the interpolation points to lie within the sampled range, and then `unique`ing the points to make sure we don't have duplicates.  

This is an ugly hack, and it would be nice to correct it in the future, but I don't think we need to do that right here.

Reply:
ok. if it's on the programmer then i'm fine with a rather evasive error message like "redundant sample positions in fast mellin transform"
if a user ever encounters that error, that user will eventually raise an issue and we'll work from there

on the other hand, if the responsability was on the user, we would have needed a much more informative error message, with recommendations on over-sampling, etc.

Reply:
Agreed.  I think this is how it wound up as an assert in the first place.